lib/quiet_quality/version_control_systems/git.rb

```diff
@@ -41,6 +61,45 @@ def self.default_branch(remote:)
       def comparison_base(sha:, comparison_branch:)
         git.merge_base(comparison_branch, sha).first.sha
       end
+
+      private def changed_lines_for(diff)
+        GitDiffParser.parse(diff).flat_map do |parsed_diff|
+          parsed_diff.changed_line_numbers.to_set
+        end
+      end
+
+      private def committed_changes(base, sha)
+        ChangedFiles.new(committed_changed_files(base, sha))
+      end
+
+      private def committed_changed_files(base, sha)
```

Comment:
Awkward naming here (which I've also been running into) - the method named like `_changed_files` is the one that does _not_ return a ChangedFiles object. The problem with my naming the data classes a bit too naturally -.-